### ISSUE TITLE
Stratis UUIDs were each claimed by more than one Stratis device

### ISSUE THREAD

**Issue by trgill** [stratis-storage/stratisd#1322]
Stratis doesn't start the pool if the pool is based on a DM linear that was created at the start of a real block device.

If I create a DM linear over the first 1 GiB of /dev/sdb - then create a pool with the linear and restart stratisd:

WARN libstratis::engine::strat_engine::engine: no pool set up, reason: Engine(Error, "failed to set up pool for (pool UUID: c28eba91-01e0-4b16-956f-d44ac7be1d98, devnodes: /dev/sdb ,/dev/dm-0): reason: Engine(Invalid, \"The following list of Stratis UUIDs were each claimed by more than one Stratis device: [Uuid { bytes: [232, 115, 221, 20, 26, 219, 68, 1, 146, 37, 236, 14, 231, 108, 33, 189] }]\")")

We should either deal with the situation or not allow the pool to be created.

**Comment by mulkieran** [comment]
Can you identify what devices the Stratis UUID belongs to?

**Comment by mulkieran** [comment]
If I understand the sequence of events, you:

1. Created a linear device mapper device on the first 1 GiB of a pool.
2. Constructed a pool using that linear device.
3. Brought stratisd down and restarted stratisd and then encountered the error.

?

The hypothesis is that stratisd identifies two block devices with the same Stratis-set device UUID: the underlying block device and then, also the DM linear device?

**Comment by trgill** [comment]
@mulkieran yes

**Comment by mulkieran** [comment]
I doubt it's something that we would be entitled to prevent. To deal w/ it requires investigating the two devices to see if they are in that relationship: DM devices laid down exactly at start of underlying device. What about a DM device aligned w/ a DM device on top of some other device? There is potentially a stack of unbounded depth.

**Comment by agrover** [comment]
Yes it's recursive. See https://github.com/agrover/melvin/blob/master/src/dm/mod.rs#L200 for an example. This was to prevent someone from adding a PV to a VG when the blockdev was e.g. a LV within the VG. Given loopback devices on a filesystem, Stratis may need to make a similar check?

But, I haven't thought about it enough to say whether this would also resolve Todd's issue with overlapping blockdevs.
